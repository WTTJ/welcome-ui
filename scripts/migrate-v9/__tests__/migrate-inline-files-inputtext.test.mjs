import { readFileSync } from 'fs'
import { resolve } from 'path'

import { describe, expect, it } from 'vitest'

import { setupMigrationTest } from '../helpers/test-setup.mjs'
import { migrateAll } from '../index.mjs'

describe('Inline Migration - InputText Components', () => {
  const tempDir = setupMigrationTest('inline-inputtext-test', '__fixtures__')

  it('migrates InputText components with styling props to InputText with classNames', async () => {
    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'InputTextComponent.tsx'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('inputtext-component.tsx')
  })
})
