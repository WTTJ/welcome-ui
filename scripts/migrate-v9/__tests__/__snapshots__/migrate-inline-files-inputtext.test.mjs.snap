// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inline Migration - InputText Components > migrates InputText components with form props to InputText with classNames > inputtext-component.tsx 1`] = `
"import { InputText } from 'welcome-ui/InputText'

const register = (name: string, options?: object) => {
  return { name, ...options }
}

export const InputTextComponent = () => {
  return (
    <form>
      <InputText
        autoFocus
        dataTestId="wa-forgot-input-email"
        name="email"
        {...register('email', {
          pattern: 'This field has invalid format',
          required: 'This field is required',
          value: /fish/,
        })}
      />
    </form>
  )
}
"
`;
